packages/core/contracts/chainbridge/SourceGovernor.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/implementation/Constants.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IBridge.sol";
+
+/**
+ * @title Governor contract on L2 that receives governance actions from L1.
+ */
+contract SourceGovernor is Ownable {
+    FinderInterface public finder;
+    uint8 public currentChainId;
+    bytes32 internal currentRequestHash;
+
+    event RelayedGovernanceRequest(
+        uint8 indexed destinationChainId,
+        address indexed to,
+        uint256 value,
+        bytes indexed data
+    );
+
+    constructor(FinderInterface _finder, uint8 _currentChainId) {
+        finder = _finder;
+        currentChainId = _currentChainId;
+        currentRequestHash = bytes32(0);
+    }
+
+    /**
+     * @notice This is the first method that should be called in order to relay a governance request to another network
+     * marked by `sinkChainID`. Note: this can only be called by the owner (presumably the L1 governor).
+     */
+    function relayGovernance(
+        uint8 destinationChainId,
+        address to,
+        uint256 value,
+        bytes memory data
+    ) external onlyOwner {
+        require(currentRequestHash == bytes32(0), "Request hash already set");
+        currentRequestHash = _computeRequestHash(to, value, data);
+        _getBridge().deposit(destinationChainId, getResourceId(), _formatMetadata(to, value, data));
+        currentRequestHash = bytes32(0);
+        emit RelayedGovernanceRequest(destinationChainId, to, value, data);
```

Comment:
would it be useful to emit the `currentRequestHash` as well?

Reply:
It's really only used internally to track the active request, so I'm not sure what use it would provide to an external observer. Thoughts?

Reply:
Yeah i don't see any use to storing it